### PR TITLE
Deblending exploration in the HSC-COSMOS field

### PR DESCRIPTION
**Pull request by jeffcarlin**
We loaded the Stack catalogs of HSC data overlapping the COSMOS fields. We show some resources for learning about the deblender and how it has identified sources. This basic framework for comparing the COSMOS catalog to HSC detections could be extended to visualize how well the Stack identifies sources in the COSMOS data (assuming COSMOS is a "truth" table).

 #Please enter the commit message for your changes. Lines starting

### CODE REVIEW

File: deblend_test.ipynb [hunk 1]
```diff
@@ -0,0 +1,1161 @@
...
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import glob\n",
+    "from collections import OrderedDict\n",
+    "import importlib\n",
+    "\n",
+    "# LSST stack imports\n",
+    "import lsst.daf.persistence as dafPersist\n",
+    "import lsst.analysis.deblender as deblender\n",
```
Comment (bstalder) [inline deblend_test.ipynb:66]: call this manual setup out!

File: deblend_test.ipynb [hunk 2]
```diff
@@ -0,0 +1,1161 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the HDF5 file of COSMOS data created in this notebook: \n",
+    "# https://github.com/lsst-com/notebooks/blob/master/hsc_cosmos_match.ipynb\n",
+    "outfile_cosmos='/project/shared/data/COSMOS_catalogs/cosmos_acs_iphot_200709.h5'\n",
+    "df_cosmos = pd.read_hdf(outfile_cosmos, 'df')"
```
Comment (bstalder) [inline deblend_test.ipynb:891]: also call out that you need tables!?!

**Comment by cwwalter** [comment]
- Note how to get lsst.analysis package.

I wasn't able to run but it all looks reasonable and I don't see any obvious problems.

**Review by cwwalter — APPROVED** [review]

File: deblend_test.ipynb [hunk 3]
```diff
@@ -0,0 +1,1161 @@
...
+   "execution_count": 52,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Picking a single parent by specifying an RA, Dec and finding the closest parent to those coords.\n",
+    "# ...so we can compare its family to COSMOS:\n",
+    "desired_ra = 150.2229\n",
+    "desired_dec = 2.25596\n",
+    "total_sep = lambda target_ra, target_dec, candidate_ra, candidate_dec: ((target_ra - candidate_ra)*np.cos(np.radians(target_dec))) ** 2.0 + (target_dec - candidate_dec)**2.0\n",
+    "pick_parent_ind = np.argmin([total_sep(desired_ra, desired_dec, pra[i], pdec[i]) for i in range(len(pra))])\n"
```
Comment (bstalder) [inline deblend_test.ipynb:1012]: variable pra is not initialized before you set an element.

**Comment by bstalder** [comment]
This notebook is on comparing the deblending between HSC Deep Coadd and COSMOS.  It successfully runs through (after installing the necessary modules). Only suggestion is that the last (money) plot is ok, but maybe having the sizes of the circles represent brightness might be more illuminating...

**Pull request merged**